lib/core/src/Cardano/Wallet/Primitive/Slotting.hs

```diff
@@ -609,3 +619,32 @@ unsafeExtendSafeZone = f . neverFails r
         , handleResult = h
         }
     r = "unsafeExtendSafeZone should make PastHorizonExceptions impossible."
+
+{-------------------------------------------------------------------------------
+                       For testing
+-------------------------------------------------------------------------------}
+
+-- | A dummy 'TimeInterpreter' for testing. A horizon at @SlotNo 40@ allows
+-- testing 'PastHorizonException's.
+dummyForkInterpreter :: TimeInterpreter (Either PastHorizonException)
+dummyForkInterpreter =
+    let
+        t0 = HF.initBound
+        t1 = HF.Bound
+                (RelativeTime 20)
+                (SlotNo 20)
+                (Cardano.EpochNo 1)
+        t2 = HF.Bound
+                (RelativeTime 40)
+                (SlotNo 40)
+                (Cardano.EpochNo 2)
+
+        era1Params = HF.defaultEraParams (SecurityParam 2) (mkSlotLength 1)
+        summary = HF.summaryWithExactly $ exactlyTwo
+            (HF.EraSummary t0 (HF.EraEnd t1) era1Params)
+            (HF.EraSummary t1 (HF.EraEnd t2) era1Params)
+        in
+            hoistTimeInterpreter (runIdentity . runExceptT) $ mkTimeInterpreter
+                nullTracer
+                (StartTime $ posixSecondsToUTCTime 0)
+                (pure $ HF.mkInterpreter summary)
```

Comment:
On reflection, I do think that this `dummyForkInterpreter` should be located together with the test code that uses it.

What do you think?

Justification:
- it's never used by the implementation code
- it's only used by test code
- the values used within are essentially arbitrary / hand-picked constants that should never be used in a real implementation.

Reply:
Fair enough! Will give moving a shot.